optimum/graphcore/trainer.py

```diff
@@ -568,7 +573,7 @@ def get_eval_dataloader(self, eval_dataset: Optional[Dataset] = None) -> poptorc
 
         # self.data_collator can be a data collator that was wrapped by pad_on_batch_axis, retrieving the original
         # data collator as the wrapped version is only wanted for training.
-        data_collator = getattr(self.data_collator, "__wrapped__", self.data_collator)
+        data_collator = getattr(self.eval_data_collator, "__wrapped__", self.data_collator)
```

Comment:
Not sure, this is intended, so double checking?
```suggestion
        data_collator = getattr(self.eval_data_collator, "__wrapped__", self.eval_data_collator)
```

Reply:
Hi James, not sure I know enough Pytorch to understand what you are talking about.
You mean you want to allow the user to specify a different batch size for training and eval? How would this affect this part of the code?

Reply:
self.data_collator can be a data collator that was wrapped by pad_on_batch_axis.
If no eval_data_collator is specified then use the unwrapped training data collator.

I suggest this change:

```
diff --git a/optimum/graphcore/trainer.py b/optimum/graphcore/trainer.py
index aeaf3b7..5297c1d 100644
--- a/optimum/graphcore/trainer.py
+++ b/optimum/graphcore/trainer.py
@@ -189,7 +189,9 @@ class IPUTrainer:
 
         default_collator = default_data_collator if tokenizer is None else DataCollatorWithPadding(tokenizer)
         self.data_collator = data_collator if data_collator is not None else default_collator
-        self.eval_data_collator = eval_data_collator if eval_data_collator is not None else self.data_collator
+        # self.data_collator can be a data collator that was wrapped by pad_on_batch_axis.
+        # If no eval_data_collator is specified then use the unwrapped training data collator.
+        self.eval_data_collator = eval_data_collator if eval_data_collator is not None else getattr(self.data_collator, "__wrapped__", self.data_collator)
         self.train_dataset = train_dataset
         self.eval_dataset = eval_dataset
         self.tokenizer = tokenizer
@@ -571,9 +573,7 @@ class IPUTrainer:
         if is_datasets_available() and isinstance(eval_dataset, datasets.Dataset):
             eval_dataset = self._remove_unused_columns(eval_dataset, description="evaluation")
 
-        # self.data_collator can be a data collator that was wrapped by pad_on_batch_axis, retrieving the original
-        # data collator as the wrapped version is only wanted for training.
-        data_collator = getattr(self.eval_data_collator, "__wrapped__", self.data_collator)
+        data_collator = self.eval_data_collator
 
         if isinstance(eval_dataset, torch.utils.data.IterableDataset):
             return poptorch.DataLoader(
@@ -620,9 +620,7 @@ class IPUTrainer:
         if is_datasets_available() and isinstance(test_dataset, datasets.Dataset):
             test_dataset = self._remove_unused_columns(test_dataset, description="test")
 
-        # self.data_collator can be a data collator that was wrapped by pad_on_batch_axis, retrieving the original
-        # data collator as the wrapped version is only wanted for training.
-        data_collator = getattr(self.data_collator, "__wrapped__", self.data_collator)
+        data_collator = self.eval_data_collator
 
         if isinstance(test_dataset, torch.utils.data.IterableDataset):
             return poptorch.DataLoader(
```